src/apiCalls.js

```diff
@@ -18,11 +20,29 @@ function getSpecificCustomer(id) {
   .then(response => response.json())
 }
 
+function postBooking(booking) {
+  return fetch('http://localhost:3001/api/v1/bookings', {
+    method: "POST",
+    body: JSON.stringify(booking),
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  })
+  .then(response => {
+    if(!response.ok) {
+      throw new Error(response.status);
+    } else {
+      resolvePost();
+    }
+  })
+  .catch(error => console.log(error));
+}
```

Comment:
Something to consider for a refactor is that this could all be one function that takes in the parameters of (path, method, body) and executes any of your fetch requests. I'm happy to show you in more detail but it would look like this:

path = the end of the url you want to fetch from (ex. customers/10)
method = type of fetch (ex. "GET")
body = object for posting (ex. {"userID": user.id, "date": selectedDate, "roomNumber": roomNumber})
Note: the body is not necessary for a GET request and it will still process fine without one so not including a body for a GET is totally fine and will still work for this function


function fetchData(path, method, body) {
    return fetch(`http://localhost:3001/api/v1/${path}`, {
        method: method,
        body: JSON.stringify(body),
        headers: {"Content-Type": "application/json"}
    })
    .then(response => {
        if(!response.ok) {
            throw new Error(response.status);
        } else {
            return response.json();
        }
    })
    .catch(error => console.log(`Fetch was unsuccessful for *${path}* due to ${error}!`));
}